jvm/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala

```diff
@@ -419,20 +495,28 @@ abstract class HashJoinLikeExecTransformer(leftKeys: Seq[Expression],
       "Join keys from two sides should have same types")
     val lkeys = HashJoin.rewriteKeyExpr(leftKeys)
     val rkeys = HashJoin.rewriteKeyExpr(rightKeys)
-    buildSide match {
+    joinBuildSide match {
       case BuildLeft => (lkeys, rkeys)
       case BuildRight => (rkeys, lkeys)
     }
   }
 
+  /**
+   * Returns whether a workaround for Anti join is needed.
+   * True for 'not exists' semantics on Velox backend.
+   */
+  def antiJoinWorkaroundNeeded: Boolean = false
+
   // Direct output order of substrait join operation
-  private val substraitJoinType = joinType match {
+  protected val substraitJoinType: JoinRel.JoinType = joinType match {
     case Inner =>
       JoinRel.JoinType.JOIN_TYPE_INNER
     case FullOuter =>
       JoinRel.JoinType.JOIN_TYPE_OUTER
-    case LeftOuter | RightOuter =>
+    case LeftOuter =>
       JoinRel.JoinType.JOIN_TYPE_LEFT
+    case RightOuter =>
+      JoinRel.JoinType.JOIN_TYPE_RIGHT
```

Comment:
Could you explain this modification? Our previous logic here was that Spark will build left for right outer join, but substrait plan always require the right side to build hash table. We need to exchange the order of left and right relation when generating the substrait plan, and the build type also need to be changed.

Reply:
Got it, thanks for your comment. I thought this was a mistake, and will change it back and add some notes here.